markdownreveal/commands.py

```diff
@@ -84,31 +94,31 @@ def upload(markdown_file: Path):
         shell('git status')
     except CalledProcessError as error:
         return
-    origin = shell('git config remote.origin.url')[0]
-    if 'github' not in origin:
-        error = 'Uploading only supported for GitHub repositories!'
+
+    if is_valid_remote(remote):
+        # We copy the directory because `git` does not follow symlinks...
+        with TemporaryDirectory() as tmpdir:
+            tmpdir = Path(tmpdir) / 'out'
+            generate(markdown_file)
+            config = load_config()
+            copytree(src=str(config['output_path']), dst=str(tmpdir))
+
+            worktree = '--work-tree=%s' % tmpdir
+            current = shell('git rev-parse --abbrev-ref HEAD')[0]
+            shell('git checkout -B gh-pages')
+            shell('git %s add .' % worktree)
+            shell('git %s commit -m "Markdownreveal live presentation"' % worktree)
+            shell('git %s push -f -u origin gh-pages' % (worktree, remote))
+            shell('git %s checkout %s' % (worktree, current))
+
+        repo = Path(origin.split(':')[-1])
```

Comment:
This `origin` variable was previously defined to be the remote URL. Now it is simply a string. Maybe you can rename this variable to be `remote_url` and set it to:

```python
remote_url = shell('git remote get-url ' + remote)[0]
```

This way the user will be notified with the correct URL the presentation was uploaded to.